helm-charts/splunk-otel-collector/templates/clusterRole.yaml

```diff
@@ -86,6 +86,14 @@ rules:
   - get
   - list
   - watch
+{{- if eq (include "splunk-otel-collector.clusterReceiverNodeDiscovererInitContainerEnabled" .) "true" }}
```

Comment:
This is probably the only place where usage of `clusterReceiverNodeDiscovererInitContainerEnabled` is needed. Both clusterReceiver template files already have `(toString $clusterReceiver.enabled) "true") (eq (toString $o11yMetricsEnabled) "true")` condition on top. So only `eq (include "splunk-otel-collector.distribution" .) "eks/fargate"` can be used there for better readability.